idea_town/frontend/static-src/test/lib/router.js

```diff
@@ -0,0 +1,100 @@
+import test from 'tape';
+import Router from '../../app/lib/router';
+import app from 'ampersand-app';
+import Me from '../../app/models/me';
+import Experiments from '../../app/collections/experiments';
+
+test(`🖳  Running Tests for ${__filename} ␄`, a => a.end());
+
+test('landing redirects correctly if logged in with addon', t => {
+  t.plan(2);
+
+  app.me = new Me({
+    user: {
+      id: 'gary@busey.biz',
+        addon: {url: '/lol nothing matters'}
+      }
+  });
+
+  app.me.hasAddon = false;
+  app.router = new Router();
+  app.router.history.start();
+
+  app.on('router:new-page', evt => {
+    if (evt.page === 'landing') {
+      t.ok(evt.page, 'Should redirect to landing if missing addon or creds');
+        app.me.hasAddon = true;
+        app.router.landing()
+      }
+
+    if (evt.page === 'home') {
+      t.ok(evt.page, 'Should redirect to home if logged in with creds');
+      app.reset();
```

Comment:
If `app.reset` needs to be called after every test completes, you'll need to make sure it gets called even if the test fails or throws. For this reason, it's a good idea to extract it out to a test runner.